### PR TITLE
Update User Guide

### LINKED ISSUE #88: Update UG in preparation for dry run
- Update the v1.3 user guide to match the current version of the product. Reason: testers will need to refer to the UG during the practical exam dry run.
- Clearly indicate which features are not implemented yet e.g. tag those features with a Coming soon.
- For those features already implemented, ensure their descriptions match the exact behavior of the product e.g. replace mockups with actual screenshots
- Landing page (docs/index.md): Update to look like a real product (rather than a project for learning SE) if you haven't done so already. In particular, update the Ui.png to match the current product

### PR DESCRIPTION
**Pull request by tsh22**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
-layout: page
-title: User Guide
+layout: page title: User Guide
 ---
 
 _Teaching Assistant_ is a desktop application that helps **JC/Secondary school teachers** have an efficient tool to keep
-track of their **schedules, contacts and tasks easily.** While it has a GUI, most of the user interactions rely on typing commands as user inputs.
+track of their **contacts and entries like tasks or schedules easily.** While it has a GUI, most of the user
+interactions rely on typing commands as user inputs.
```
Comment (goatygoatygoat) [inline docs/UserGuide.md]: I wrote a small intro last time:

Welcome to the User Guide of **Teaching Assistant**!

Are you a JC/Secondary school teacher, having troubles keeping track of all your consultations, meetings and your students' contacts? No worries! Our application, **Teaching Assistant** will provide an all-in-one platform for you to organise your entries _(tasks and schedules)_ and contacts!

Teaching Assistant mainly uses a Command Line Interface (CLI). This means that a user can use the application by typing into a Command Box. For users who type fast, they can use this application more efficiently than other applications that heavily use Graphical User Interface (GUI), where a user uses the application by interacting with graphical features such as buttons.

If you are interested, jump to [Quick Start](#quick-start) to learn how to learn how to start using **Teaching Assistant**.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ This is a mockup of our UI.
 - [Features](#features)
     - [Viewing help](#viewing-help)
     - [Contact](#adding-a-contact)
```
Comment (goatygoatygoat) [inline docs/UserGuide.md:29]: Can add hyperlinks to each command under the Contact category, same for Entry category.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -35,129 +35,237 @@ This is a mockup of our UI.
```
Comment (goatygoatygoat) [inline docs/UserGuide.md:62]: Can add screenshots from our demo under each feature (but is optional for now)

File: docs/UserGuide.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-title: User Guide
+layout: page title: User Guide
 ---
 
 _Teaching Assistant_ is a desktop application that helps **JC/Secondary school teachers** have an efficient tool to keep
-track of their **schedules, contacts and tasks easily.** While it has a GUI, most of the user interactions rely on typing commands as user inputs.
+track of their **contacts and entries like tasks or schedules easily.** While it has a GUI, most of the user
+interactions rely on typing commands as user inputs.
 
 This is a mockup of our UI.
```
Comment (goatygoatygoat) [inline docs/UserGuide.md:21]: Change the pic to our actual Ui.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -35,129 +35,237 @@ This is a mockup of our UI.
...
 Finds an existing contact by name in the address book.
 
 Format: `find KEYWORD [MORE_KEYWORDS]`
-* The search is case-insensitive e.g. amy will match Amy
+
+* Only names are searched.
+* The search is case-insensitive e.g. `amy` will match `Amy`.
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only full words will be matched e.g. `Han` will not match `Hans`.
+* Persons matching at least one keyword will be returned (i.e. OR search). E.g. `Hans Bo` will return `Hans Gruber`
```
Comment (goatygoatygoat) [inline docs/UserGuide.md]: I've changed the implementation such that `Hans Bo` will only return `Hans Bo` (i.e. AND search)!

File: docs/UserGuide.md [hunk 6]
```diff
@@ -35,129 +35,237 @@ This is a mockup of our UI.
...
+  , `Bo Yang`.
+
+Example(s):
+
+* `find John` returns `john` and `John Doe`
+* `find alex david` returns `Alex Yeoh`, `David Li`
```
Comment (goatygoatygoat) [inline docs/UserGuide.md]: Will not return anything unless there is a person called `alex david`

**Review by goatygoatygoat — CHANGES_REQUESTED** [review]

**Review by lrj689 — APPROVED** [review]

File: docs/UserGuide.md [hunk 7]
```diff
@@ -16,13 +16,13 @@ This is a mockup of our UI.
...
+    - [Entry](#adding-an-entry)
 - [Command Summary](#command-summary)
 
 ---
 
 ## Quick Start
+
 1. Ensure you have Java 11 installed in your computer.
 1. Download the latest `teachingAssistant.jar` [here](https://github.com/AY2021S2-CS2103T-W13-4/tp/releases).
 1. Copy the file to the folder you want to use as the *home folder* for your AddressBook.
```
Comment (goatygoatygoat) [inline docs/UserGuide.md]: for your Teaching Assistant

**Review by goatygoatygoat — APPROVED** [review]

**Review by KevinLohJunYong — APPROVED** [review]

**Pull request merged**